### ISSUE TITLE
Safe use of deferred jQuery even if there are inline scripts depending on it

### ISSUE THREAD

**Issue by DahmaniAdame** [wp-media/wp-rocket#3637]
**Is your feature request related to a problem? Please describe.**
When jQuery is deferred, inline scripts will report a missing dependency and fail to execute properly.

**Describe the solution you'd like**
It looks like deferred scripts are executed right before DOMContentLoaded.

We can wrap jQuery-dependent inline scripts inside a listener to fire only after DOMContentLoaded starts. 

That way, we keep jQuery deferred without risking that inline scripts will fail. 

This should help eliminate render-blocking notices resulting from excluding jQuery from deferring. 

Example listener: 

```
window.addEventListener('DOMContentLoaded', (event) => {
//  jQuery scripts goes here
});
```

We can regex jQuery and $ to be wrapped inside the listener while we process the buffer. 

**Describe alternatives you've considered**
N/A

**Additional context**
Reference about defer scripts being fully loaded/executed before DOMContentLoaded - https://javascript.info/script-async-defer

**Comment by DahmaniAdame** [comment]
Looks like we are doing that right here - https://github.com/wp-media/wp-rocket/blob/05fe10672badbd6d0e900c11a61ed74041e74487/tests/Fixtures/inc/Engine/Optimization/DeferJS/Subscriber/deferInlineJs.php
